### PR TITLE
resolves #36: .setYear() .setMonth() .setDay() methods added

### LINKED ISSUE #36: implement setters for date components
Methods to be added
- .setYear()
- .setMonth()
- .setDay()

### PR DESCRIPTION
**Pull request by thebishalniroula**
### Description
.setYear()
.setMonth()
.setDay() methods added

### Issue Reference
implement setters for date components #36

### Checklist

- [x] Lint rules are passed correctly
- [x] Code builds correctly
- [x] Created or updated tests for the new code
- [x] All existing and new tests passed
- [x] Pull request title is concise and descriptive.
- [x] Pull request is linked to the related issue.

### CODE REVIEW

File: src/BikramSambat.ts [hunk 1]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
     }
   }
 
+  /**
+   * Sets the year property to the provided value.
+   * @param {number | undefined} year
+   */
+  public setYear(year: number | undefined) {
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
  public setYear(year: number) {
```

File: src/BikramSambat.ts [hunk 2]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
...
+    this.year = year
+  }
+
+  /**
+   * Sets the month property to the provided value.
+   * @param {number | undefined} month
+   */
+  public setMonth(month: number | undefined) {
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
  public setMonth(month: number) {
```

File: src/BikramSambat.ts [hunk 3]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
...
+    this.month = month
+  }
+
+  /**
+   * Sets the day property to the provided value.
+   * @param {number | undefined} day
+   */
+  public setDay(day: number | undefined) {
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
  public setDay(day: number) {
```

File: src/BikramSambat.ts [hunk 4]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
     }
   }
 
+  /**
+   * Sets the year property to the provided value.
+   * @param {number | undefined} year
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
   * @param {number} year
```

File: src/BikramSambat.ts [hunk 5]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
...
+   */
+  public setYear(year: number | undefined) {
+    this.year = year
+  }
+
+  /**
+   * Sets the month property to the provided value.
+   * @param {number | undefined} month
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
   * @param {number} month
```

File: src/BikramSambat.ts [hunk 6]
```diff
@@ -56,6 +56,30 @@ export default class BikramSambat {
...
+   */
+  public setMonth(month: number | undefined) {
+    this.month = month
+  }
+
+  /**
+   * Sets the day property to the provided value.
+   * @param {number | undefined} day
```
Comment (blackkspydo) [inline src/BikramSambat.ts]:
```suggestion
   * @param {number} day
```

**Review by blackkspydo — CHANGES_REQUESTED** [review]
I don't think we should allow undefined as argument for setters.

**Comment by thebishalniroula** [comment]
@blackkspydo Fixed.

**Review by blackkspydo — APPROVED** [review]
LGTM

**Pull request merged**